inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/file/FileNameRegexMatchPredicate.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.utils.file;
+
+import com.google.common.base.Predicate;
+
+import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileNameRegexMatchPredicate implements Predicate<File> {
+
+    private Pattern pattern;
+
+    public FileNameRegexMatchPredicate(String regex, boolean caseSensitive) {
+        if (!caseSensitive) {
+            pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+        } else {
+            pattern = Pattern.compile(regex);
+        }
+    }
+
+    public boolean apply(File input) {
+        Matcher matcher = pattern.matcher(input.getAbsolutePath());
+        if (matcher.matches()) {
+            return true;
+        } else {
+            if (matcher.lookingAt()) {
```

Comment:
can simplify to return matcher.lookingAt()

Reply:
ok